Noaa demo

cawasser opened the pull request:
Updates to add "telemetry" data (service and widget) to the baseline. Some changes were made to carousel.cljs to better process :line-chart in a carousel, used by both spectrum and telemetry.

cawasser committed:
- added code to activate the "first" layer on the WW map
- updated heatmap-service to provide something more like a Mercator-projection of data-rate

cawasser committed:
- basic heatmap implemented for a subset of all the epochs

cawasser committed:
- carousel added for heatmap widget (update to vanilla_db, but not a newsource)

cawasser committed:
- removed "add widget" button form carousel (abandoned feature)

cawasser committed:
- switched "spectrum traces" to a carousel
- removed "Add Carousel" button (obe, avoids confusion)
- another vanilla_db update

cawasser committed:
- performance tweaks? turned off reframe-10x, removed :categories form heatmap-data, some changed to WW widget handling of r/atoms

cawasser committed:
- refactored to (mount-components) so we can tell sahdow-cljs to run the function on a hot reload the UI using ^:dev/after-load-async. not perfect, but the UI now does update during hot-reload-workflow

cawasser committed:
- most of a new "telemetry service" is working, just need to sort out handling Datetime coming form server for visualization
- NEW VANILLA_DB!!!!!

cawasser committed:
- getting some data to display on telemetry carousel

cawasser committed:
- moved axis labels into the data sent by the server (spectrum AND telemetry carousels)
- refactored (big-chart) to work across spectrum and telemetry carousels
- returning ALL telemetry data

cawasser committed:
- updated per dependabot recommendations:
   - Bumps elliptic from 6.5.2 to 6.5.3
   - Bumps lodash from 4.17.15 to 4.17.19

cawasser committed:
- adding the source data

KiraFarney5 commented on src/clj/vanilla/heatmap_service.clj:
```diff
@@ -1,17 +1,37 @@
+    [
+     ;{:name        "Fruit Production per Continent"
+     ; :keys ["x" "y" "value"]
+     ; :categories {:x ["Apples" "Avocados" "Bananas" "Oranges" "Peaches" "Pears" "Plums" "Prunes" "Starfruit" "Tangerine"]
```
This is the yummiest line of code!

cawasser committed:
- add "type" for each axis on (big-chart) so we get dates handled properly (without any conversion form #inst on the CLJS side)
- provide a handful of the 29 metric in the spreadsheet (still pretty slow on the client-side)

cawasser committed:
- removed unused datetime conversions form timeline.cljs

austinpaine commented on src/cljs/vanilla/widgets/carousel.cljs:
```diff
@@ -147,96 +149,206 @@
+
+
+
+(defn make-telemetry-widget [name data options]
```
Do we want to genericize these carousel widget functions?  I know we need to re-architect the overall widget/carousel creation structure, but in the meantime, this is a lot of duplicate code.   It does seem like theres just enough differences in sources and titles and such between them to justify, just thinking.

cawasser replied:
I'm pretty sure widget-workshop will sort this all out since it's a design goal to do just that, among many other things about widgets and data-sources, but this repetition shows that such a re-think is really, really necessary.

austinpaine commented on src/cljs/vanilla/mapping/layer_management.cljs:
```diff
@@ -168,12 +170,17 @@
+
+      (for [[idx e] (map-indexed vector (reverse (sort epochs)))]
+        (do
+          (prn "idx " idx)
```
commented this and line 159 out to clear up console

austinpaine commented on src/clj/vanilla/db/core.clj:
```diff
@@ -133,9 +133,9 @@
-      ; "returns carousel widget that supports multiple widgets"]
+      ["24000" "telemetry-service" "Telemetry"
+       "data-format/grouped-x-y" "vanilla.telemetry-service/fetch-data"
+       "returns carousel widget of spave vehicel telemetry data"]
```
fixed this typo

austinpaine reviewed (approved):
Looks good to me, the tweaks to keep data amount reasonable seemed to help, way less crashes.  I pushed a tiny commit to fix a typo and two things I had already commented out.

Commit:
typo in db.core, two comments, ww navbar title key warning fix

chadarm reviewed (approved):
Looks ready to go.
